LANDSCAPE/COMPARISONS/ARGOCD-FLUX/index.md

```diff
@@ -0,0 +1,275 @@
+
+
+## Argo CD
+
+<span style="text-decoration:underline;">Raw Kubernetes YAML:</span>
+
+Argo CD also supports Kustomize and raw Kubernetes YAML. It can, however, apply raw YAML files directly, without using Kustomize. This can have side effects of Argo CD Applications not syncing right aways as resources need to be reapplied in a retry ([which can be set up per Application](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/application.yaml#L111-L116)), or use ordering built into Argo CD. Argo CD does the equivalent of `kubectl apply -f` or a `kustomzie build` if Kustomize is detected. This is by design as the Argo CD creators wanted to emulate the `kubectl` client as closely as possible.
+
+<span style="text-decoration:underline;">Helm:</span>
+
+Argo CD also has support for Helm, but approaches it in a different way. Argo CD does not use the Helm golang library directly, but instead, uses helm to render the template files and pipes them directly to `kubectl`. Helm chart hooks are converted into Argo CD Syncwaves/Hooks (more on this next) where it can.This has the side effect of users not being able to use the `helm` cli against the cluster because `helm install` wasn’t called in the first place (i.e. `helm ls `will return empty). 
+
+<span style="text-decoration:underline;">Things to Note:</span>
+
+Argo CD can do “auto discovery” with Kustomize like Flux does by setting it in the sync settings. This isn’t the default, but can be configured by the user
+
+With Argo CD you have more control over ordering of individual resources (i.e. specific YAML file) with [Syncwaves and Phases](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/). This allows you to apply each YAML manifest one at a time. Using Syncwaves and Phases, Argo CD will wait until one YAML is applied (and healthy) before moving onto applying the next one. 
+
+There is a controller that supplements Argo CD’s Application controller called [Application Sets](https://github.com/argoproj-labs/applicationset). This add-on lets users deploy many Applications across one or many managed clusters.
+
+
+# Secrets Management
+
+GitOps practice of storing the source of truth in git, has had some contention with respect to storing [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/). This has to do with the fact that Secrets often contain sensitive 	information like passwords or tokens, and these secrets are only encoded.
+
+Both Flux and Argo CD have mechanisms in place to handle the encrypting of secrets.
+
+
+## FluxCD
+
+FluxCD provides a guide for managing encrypted secrets with [Mozilla SOPS](https://fluxcd.io/docs/guides/mozilla-sops/) or [Sealed Secrets](https://fluxcd.io/docs/guides/sealed-secrets/) so that Kubernetes secrets can be stored safely in Git, secured by any of OpenPGP, AWS KMS, GCP KMS and Azure Key Vault. Flux’s Kustomization objects natively support decrypting Mozilla SOPS secrets.
```

Comment:
I'm not sure if this is worth calling out specifically for Flux? The documentation provides guides for these methods, but Flux doesn't provide any additional value on top of these.

Reply:
```suggestion
Flux CD provides a guide for managing encrypted secrets with [Mozilla SOPS](https://fluxcd.io/docs/guides/mozilla-sops/) or [Sealed Secrets](https://fluxcd.io/docs/guides/sealed-secrets/) so that Kubernetes secrets can be stored safely in Git, secured by any of OpenPGP, AWS KMS, GCP KMS and Azure Key Vault. Flux’s Kustomization objects natively support decrypting Mozilla SOPS secrets.
```

Reply:
@RedbackThomson I understand what you mean because both Argo CD and Flux are unopinionated about how secrets are managed.

My hypothesis is that @christianh814 et al. are frequently asked about Secrets Management in GitOps, thus added this `# Secrets Management` to address the FAQ situation.

Reply:
Yes. We are. And we typically point to a part of the CNCF Landscape (sealed secrets, Vault, etc.).